app/Libraries/Markdown/Osu/DocumentProcessor.php

```diff
@@ -221,6 +228,38 @@ private function recordFirstImage()
         $this->firstImage = proxy_media($this->node->getUrl());
     }
 
+    private function setImageGallery()
+    {
+        if (!$this->node instanceof Inline\Image || !$this->event->isEntering()) {
+            return;
+        }
+
+        $imageClass = $this->node->data['attributes']['class'];
+        $blockClass = $this->config->get('osu_extension/block_name');
+
+        if ($imageClass !== "{$blockClass}__figure-image") {
+            return;
+        }
+
+        $imageUrl = $this->node->getUrl();
+        if (starts_with($imageUrl, route('wiki.show', [], false))) {
```

Comment:
check if doesn't start with `https?://` instead similar to the one in `proxy_media`

Reply:
Now I look at this again, can we just change `proxy_media` to this?

```diff
--- a/app/helpers.php
+++ b/app/helpers.php
@@ -922,6 +922,15 @@ function proxy_media($url)
 
     $url = html_entity_decode_better($url);
 
+    // turn relative urls into absolute urls
+    if (!preg_match('/^https?\:\/\//', $url)) {
+        // ensure url is relative to the site root
+        if ($url[0] !== '/') {
+            $url = "/{$url}";
+        }
+        $url = config('app.url').$url;
+    }
+
     if (config('osu.camo.key') === null) {
         return $url;
     }
@@ -932,15 +941,6 @@ function proxy_media($url)
         return $url;
     }
 
-    // turn relative urls into absolute urls
-    if (!preg_match('/^https?\:\/\//', $url)) {
-        // ensure url is relative to the site root
-        if ($url[0] !== '/') {
-            $url = "/{$url}";
-        }
-        $url = config('app.url').$url;
-    }
-
 
     $hexUrl = bin2hex($url);
     $secret = hash_hmac('sha1', $url, config('osu.camo.key'));
```

So it adds `app.url` prefix to absolute url when there's no camo setup.